test/e2e/nodepool_test.go

```diff
@@ -0,0 +1,55 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/onsi/gomega"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	zeroReplicas int32 = 0
+	oneReplicas  int32 = 1
+	twoReplicas  int32 = 2
+)
+
+func TestNodePool(t *testing.T) {
+	g := NewWithT(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	clusterOpts := globalOpts.DefaultClusterOptions(t)
+	defer cancel()
+
+	mgmtClient, err := e2eutil.GetClient()
+	g.Expect(err).NotTo(HaveOccurred(), "failed to get k8s client")
+
+	guestCluster := e2eutil.CreateCluster(t, ctx, mgmtClient, &clusterOpts, globalOpts.Platform, globalOpts.ArtifactDir)
+	guestClient := e2eutil.WaitForGuestClient(t, ctx, mgmtClient, guestCluster)
+
+	// Wait for the rollout to be reported complete
+	t.Logf("Waiting for cluster rollout. Image: %s", globalOpts.LatestReleaseImage)
+	e2eutil.WaitForImageRollout(t, ctx, mgmtClient, guestClient, guestCluster, globalOpts.LatestReleaseImage)
+
+	// Set of tests
+	// Each test should have their own NodePool
+	t.Run("TestNodePoolAutoRepair", testNodePoolAutoRepair(mgmtClient, guestCluster, guestClient, clusterOpts))
+}
+
+func nodePoolScaleDown(ctx context.Context, client crclient.Client, nodePool hyperv1.NodePool, g gomega.Gomega) {
```

Comment:
g gomega.Gomega, should not be a param, this func should return an error and so the consumer can check the error with g

Reply:
Otherwise you should pass t.

And you must pass t and used it for t.Errorf if you want to log here as in line 52